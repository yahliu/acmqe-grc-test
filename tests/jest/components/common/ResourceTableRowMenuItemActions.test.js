/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */

'use strict'

import { resourceActions } from '../../../../src-web/components/common/ResourceTableRowMenuItemActions'

describe('ResourceTableRowMenuItemActions', () => {
  it('table.actions.edit is called as expected', () => {
    const action = 'table.actions.edit'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.launch.cluster is called as expected', () => {
    const action = 'table.actions.launch.cluster'
    const resourceType = {
      'name': 'PoliciesByCluster',
      'query': 'ALL_POLICIES'
    }
    const data = {
      consoleURL: 'website'
    }
    window.open = jest.fn()
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, data, true, {})).toBeUndefined()
    expect(dispatch).not.toHaveBeenCalled()
    expect(window.open).toHaveBeenCalledTimes(1)
    expect(window.open.mock.calls[0]).toMatchSnapshot()
    window.open.mockClear()
  })
  it('table.actions.compliance.remove is called as expected', () => {
    const action = 'table.actions.compliance.remove'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.policy.remove is called as expected', () => {
    const action = 'table.actions.policy.remove'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.remove is called as expected', () => {
    const action = 'table.actions.remove'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.disable is called as expected', () => {
    const action = 'table.actions.disable'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.enable is called as expected', () => {
    const action = 'table.actions.enable'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.enforce is called as expected', () => {
    const action = 'table.actions.enforce'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('table.actions.inform is called as expected', () => {
    const action = 'table.actions.inform'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch.mock.calls[0]).toMatchSnapshot()
  })
  it('does nothing and returns undefined given invalid action', () => {
    const action = 'not.an.action'
    const resourceType = {
      'name': 'HCMCompliance',
      'query': 'ALL_POLICIES'
    }
    const dispatch = jest.fn()
    expect(resourceActions(action, dispatch, resourceType, {}, true, {})).toBeUndefined()
    expect(dispatch).toHaveBeenCalledTimes(0)
    expect(dispatch.mock.calls[0]).toBeUndefined()
  })
})
