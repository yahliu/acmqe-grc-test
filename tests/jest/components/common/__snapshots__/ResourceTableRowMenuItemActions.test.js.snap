// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceTableRowMenuItemActions table.actions.compliance.remove is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.remove-hcmcompliance.heading",
        "label": "modal.remove-hcmcompliance.label",
        "primaryBtn": "modal.remove-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-remove",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.disable is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.disable-hcmcompliance.heading",
        "label": "modal.disable-hcmcompliance.label",
        "primaryBtn": "modal.disable-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-disable",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.edit is called as expected 1`] = `
Array [
  "undefined/all/default/policy-pod/edit",
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.enable is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.enable-hcmcompliance.heading",
        "label": "modal.enable-hcmcompliance.label",
        "primaryBtn": "modal.enable-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-enable",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.enforce is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.enforce-hcmcompliance.heading",
        "label": "modal.enforce-hcmcompliance.label",
        "primaryBtn": "modal.enforce-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-enforce",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.inform is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.inform-hcmcompliance.heading",
        "label": "modal.inform-hcmcompliance.label",
        "primaryBtn": "modal.inform-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-inform",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.launch.cluster is called as expected 1`] = `
Array [
  "website",
  "_blank",
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.policy.remove is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.remove-hcmcompliance.heading",
        "label": "modal.remove-hcmcompliance.label",
        "primaryBtn": "modal.remove-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-remove",
    },
    "type": "MODAL_UPDATE",
  },
]
`;

exports[`ResourceTableRowMenuItemActions table.actions.remove is called as expected 1`] = `
Array [
  Object {
    "data": Object {
      "data": Object {
        "apiVersion": undefined,
        "kind": "HCMCompliance",
      },
      "label": Object {
        "heading": "modal.remove-hcmcompliance.heading",
        "label": "modal.remove-hcmcompliance.label",
        "primaryBtn": "modal.remove-hcmcompliance.primaryBtn",
      },
      "open": true,
      "resourceType": Object {
        "name": "HCMCompliance",
        "query": "ALL_POLICIES",
      },
      "type": "resource-remove",
    },
    "type": "MODAL_UPDATE",
  },
]
`;
